scripts/auto_archive.py

```diff
@@ -0,0 +1,202 @@
+    os.makedirs(os.path.dirname(out_dir), mode=0o755, exist_ok=True)
+    out_filename = f"{out_dir}_version-{version}.tar.gz"
+    logger.info(f"Archiving dataset: {dataset_path} to {out_filename}")
+
+    cwd = os.getcwd()
+    try:
+        datalad_archiver = export_archive.ExportArchive()
+        dataset_repo = git.Repo(dataset_path)
+
+        with open(os.path.join(dataset_path, ".git.log"), "w") as fout:
+            fout.write(dataset_repo.git.log(pretty="format:%H %s"))
+
+        # Export is performed from the dataset root.
+        # This is to avoid failure when a submodule is not downloaded; e.g. for parent
+        # dataset in dataset derivative.
+        os.chdir(os.path.join(cwd, dataset_path))
+        datalad_archiver(".", filename=out_filename)
+
+    except Exception as e:
+        raise ArchiveFailed(
+            f"FAILURE: could not archive dataset: {dataset_path} to {out_filename}\n{e}"
+        )
+    finally:
+        os.chdir(cwd)
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    datasets = get_all_datasets() if args.all else get_modified_datasets()
+
+    for dataset in datasets:
+        dataset_name = dataset.removeprefix("projects/")
+
+        try:
+            logger.info(f"Installing dataset: {dataset}")
+            install_dataset(dataset)
+
+            is_public = False
+            version = ""
+            dataset_size = 0.0
+
+            with open(os.path.join(dataset, "DATS.json")) as fin:
+                metadata = json.load(fin)
+
+                is_public = metadata.get("privacy") == "open"
+                version = metadata.get("version")
+
+                for distribution in metadata.get("distributions", list()):
+                    dataset_size += humanfriendly.parse_size(
+                        f"{distribution['size']} {distribution['unit']['value']}",
+                    )
+                    dataset_size //= 1024 ** 3  # Convert to GB
+
+            # Only archive public dataset less than 20GB
+            if dataset_size <= args.max_size and is_public:
+                logger.info(f"Downloading dataset: {dataset}")
+                get_dataset(dataset)
+                for submodule in get_proper_submodules(dataset):
+                    get_dataset(submodule)
+
+                archive_dataset(
+                    dataset,
+                    out_dir=os.path.join(args.out_dir, dataset_name),
+                    version=version,
+                )
+                logger.info(f"SUCCESS: archive created for {dataset}")
+
+            else:
+                logger.info(f"SKIPPED: archive not needed for {dataset}")
+
+        except Exception as e:
+            # TODO implement notification system.
+            # This will alert when a dataset fails the archiving process.
+            logger.exception(
+                f"FAILURE: could not archive dataset: {dataset} to {args.out_dir}.tar.gz\n{e}"
```

Comment:
is there a descriptions of what went wrong that comes with the Exception when something goes wrong?

Reply:
`logger.exception` will log the given message and the stack trace from the occurring exception.
e.g. In the above case, the "FAILURE ..." mesage , the exception message and the stack trace will be logged.